### PR TITLE
Add server action 

### PR DESCRIPTION
**Pull request by singh12345akshay**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+NEXT_PUBLIC_BACKEND_URL = "http://localhost:5030/graphql"
```
Comment (SamyakJoshi) [inline .env:1]: add .env to .gitignore

File: components/todoTable/addItem.tsx [hunk 2]
```diff
@@ -1,39 +1,17 @@
...
-    } catch (error) {
-      console.log("Error adding todo:");
-    }
-
-    // Resetting the data.
-    setTitle("");
-    setAssignee("");
-    setDescription("");
-  };
+const reqBody={content: "task testing", description:"this task is for testing", assignee:"jaya patidar"}
```
Comment (SamyakJoshi) [inline components/todoTable/addItem.tsx:9]: can we remove this static data here

File: components/todoTable/addItem.tsx [hunk 3]
```diff
@@ -118,9 +96,9 @@ function AddItem() {
             </button>
           </div>
         </div>
-      )}
+      )} */}
```
Comment (SamyakJoshi) [inline components/todoTable/addItem.tsx:99]: remove commented code

File: components/todoTable/addItem.tsx [hunk 4]
```diff
@@ -1,39 +1,17 @@
-"use client";
 
 import { useState } from "react";
 import { useMutation } from "@apollo/client";
 import { ADD_TODO, GET_TODOS } from "../../graphql";
```
Comment (SamyakJoshi) [inline components/todoTable/addItem.tsx:4]: please fix the import it should be used with relative path

File: src/app/layout.tsx [hunk 5]
```diff
@@ -11,6 +13,19 @@ export default function RootLayout({
...
+  // const query2 = ADD_TODO;
+  // const data2 = getTodoData({
+  //   query: query2,
+  //   variables: { content: "this is new task ", description:"this task is for testing using fetch", assignee:"Akshay Prata Singh" },
+  // });
+
+  fetchGraphQl({ query });
+  // const deleteQuery = DELETE_TASK;
+  // const data = getTodoData({
+  //   query: deleteQuery,
```
Comment (SamyakJoshi) [inline src/app/layout.tsx:26]: please remove commented code

File: src/app/todo/loading.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Loading = () => {
+  return (
+    <div>Loading</div>
+  )
+}
+
+export default Loading;
```
Comment (SamyakJoshi) [inline src/app/todo/loading.tsx:9]: is this required ?

File: src/lib/actions/interface.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    assignee: string;
+    content: string;
+    description: string;
+    _id: string;
+    createdAt: string;
+    isDone: boolean;
+    sharedUsers: string;
+    startTime: string;
+    updatedAt: string;
+}
```
Comment (SamyakJoshi) [inline src/lib/actions/interface.ts:21]: we can enhance this a bit more but lets do that in another issue

File: src/lib/actions/todo.ts [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+export const getTodo = async (): Promise<ITaskData[]> => {
+    try {
+        const res = await fetchGraphQl<IRequestBody, IBaseGraphqlResponse<IAllTaskData>>({ query: GET_TODOS });
+        return res.data.getAllTasks;
+    }catch(error){
+        console.log(error)
+        throw new Error('Failed to Fetch data');
+    }
+};
```
Comment (SamyakJoshi) [inline src/lib/actions/todo.ts:34]: format properly

File: src/lib/actions/todo.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+
+
+export const addTodo = async (reqBody: AddTodoReqBody): Promise<AddTodoResponseDetail> => {
+    try {
+        const res = await fetchGraphQl<AddTodoReqBody, IBaseGraphqlResponse<AddTodoResponse>>({ query: ADD_TODO, variables: reqBody })
+        return res.data.createTodo;
+    } catch (error) {
+        console.log(error)
+        throw new Error('Failed to add todo: ' + error);
```
Comment (SamyakJoshi) [inline src/lib/actions/todo.ts:44]: dont throw a error in the catch or remove catch block

File: src/lib/actions/todo.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import { ADD_TODO, GET_TODOS } from "@/graphql";
+import { IRequestBody, fetchGraphQl } from "..";
```
Comment (BMJ-075) [inline src/lib/actions/todo.ts:2]: maybe import this using aliases or relative as they are more readable ?

File: src/lib/actions/todo.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    sharedUsers: string
+    startTime: string
+    updatedAt: string
+}
+
+interface IAllTaskData {
+    getAllTasks: ITaskData[];
+}
+
+// To fetch all
```
Comment (BMJ-075) [inline src/lib/actions/todo.ts:27]: can we make documentation ore descritptive

**Review by SamyakJoshi — APPROVED** [review]

**Pull request merged**